src/Ordering.API/Application/Queries/OrderQueries.cs

```diff
@@ -1,85 +1,57 @@
 ﻿namespace eShop.Ordering.API.Application.Queries;
 
-public class OrderQueries(NpgsqlDataSource dataSource)
+public class OrderQueries(IOrderRepository orderRepository, OrderingContext context)
     : IOrderQueries
 {
     public async Task<Order> GetOrderAsync(int id)
     {
-        using var connection = dataSource.OpenConnection();
-
-        var result = await connection.QueryAsync<dynamic>("""
-            SELECT o."Id" AS ordernumber, o."OrderDate" AS date, o."Description" AS description, o."Address_City" AS city,
-                o."Address_Country" AS country, o."Address_State" AS state, o."Address_Street" AS street,
-                o."Address_ZipCode" AS zipcode, os."Name" AS status, oi."ProductName" AS productname, oi."Units" AS units,
-                oi."UnitPrice" AS unitprice, oi."PictureUrl" AS pictureurl
-            FROM ordering.Orders AS o
-            LEFT JOIN ordering."orderItems" AS oi ON o."Id" = oi."OrderId"
-            LEFT JOIN ordering.orderstatus AS os ON o."OrderStatusId" = os."Id"
-            WHERE o."Id" = @id
-            """,
-            new { id });
-
-        if (result.AsList().Count == 0)
+        var order = await orderRepository.GetAsync(id);
```

Comment:
GetAsync in OrderRepository is implemented particularly inefficient, doing Find and then two LoadAsync for the related entities - that's 3 roundtrips where only one should be done. Given that the previous code just issued direct SQL - bypassing the repository - maybe it's OK to bypass the repository and just integrate a LINQ query here.

Note that this is what's done below in GetOrdersFromUserAsync; we should be consistent here, either always going through the repository (in which case the LINQ query in GetOrdersFromUserAsync needs to move to the repository), or not doing it (which I'd recommend at least for now).

Stepping back, there's a fundamental question here of whether we want to go use the repository abstraction or not. The general feeling in the team is that a repository abstraction over EF (IOrderRepository) isn't something that should be done by default, and makes sense specifically in the following cases:

* Multiple databases are being targeted, and LINQ queries require tweaking on a per-database basis.
* Unit tests are desired which don't utilize the production database system; the repository layer allows mocking at that point (much better than swapping EF providers with InMemory/Sqlite). However, we don't believe this should be done by default - testing against the real database with e.g. testcontainers is generally a better approach, at least as a baseline (these need to be done anyway to test that queries execute correctly against the real database system).

We should maybe discuss exactly what it is we're trying to show/achieve with eshop, but there's a good chance I'd be for removing the repository layer altogether.

Reply:
Thank you for your feedback. In my benchmarks, using the repository as the "naive approach" indeed proved to be quite slow. However, in my second branch where I integrated benchmarks, abandoning the repository led to significantly improved performance, almost matching that of raw SQL queries. So great work by the EF Core team!

For query operations, bypassing the repository seems to be a viable and efficient approach.

Additionally, I'd like to highlight another potential use case for the repository pattern. This involves retrieving aggregate roots from the database, ensuring that they are consistently and correctly fetched. Where this cannot be achieved with EF (configuration) This consistency is crucial, especially in complex domain models where an aggregate root encompasses various related entities, maintaining their invariants and business rules.

I apologize for any confusion caused. I've already addressed some issues on the benchmark branch and was awaiting further confirmation. Initially, I was hesitant to refactor extensively, but after reviewing your comments, it's clear that our perspectives align.

Moving forward, I'm inclined to consider removing the repository layer, especially given the current context and objectives with the eshop project.

Reply:
> This involves retrieving aggregate roots from the database, ensuring that they are consistently and correctly fetched.

That's interesting.. If the intent is to always fetch the related entities (which is how aggregate root is sometimes interpreted), then it probably makes more sense to model this as owned entities, or to use the auto-include feature; that ensures that whenever a principal is loaded, its dependents are loaded as well. Though it's indeed possible that when this code was written these two features weren't yet available.

In any case, I'm not sure eshop should necessarily be showcasing specific modeling patterns (that's a genuine open question - the exact purpose of this sample app is't necessarily 100% defined). I'd lean towards doing the simpler thing here rather than something very involved with patterns that may not be necessary (or even helpful) for most cases. 

> I apologize for any confusion caused.

No need to apologize at all! This is bringing the code forward. If you intend to continue working on this, I indeed recommend doing step-by-step changes for easier review/discussion rather than a big refactor all at once.

> It seems that the use of LoadAsync in the repository was implemented before the availability of AsSplitQuery

That's definitely possible - though I doubt that the performance issues of single query (e.g. cartesian explosion due to joins) was the reason this was written this way...

Reply:
Small question can I combine the refactoring of order and orderitem where I will introduce the properties (instead of the getter and setter methods) and the refactoring to LINQ or is it better to make separate PR for that change?

Reply:
Sure, as long as it's simple stuff (introduce properties instead of get/set methods), no problem.